valhalla/baldr/rapidjson_utils.h

```diff
@@ -323,103 +333,54 @@ class writer_wrapper_t {
     writer.SetMaxDecimalPlaces(precision);
   }
 
-  inline void operator()(const char* key, const char* value) {
-    writer.String(key);
-    writer.String(value);
-  }
-
-  inline void operator()(const char* key, const std::string& value) {
-    writer.String(key);
-    writer.String(value);
-  }
-
-  inline void operator()(const char* key, const double value) {
-    writer.String(key);
-    writer.Double(value);
-  }
-
-  inline void operator()(const char* key, const uint64_t value) {
-    writer.String(key);
-    writer.Uint64(value);
-  }
-
-  inline void operator()(const char* key, const int64_t value) {
-    writer.String(key);
-    writer.Int64(value);
-  }
-
-  inline void operator()(const char* key, const bool value) {
-    writer.String(key);
-    writer.Bool(value);
-  }
-
-  inline void operator()(const char* key, const std::nullptr_t) {
-    writer.String(key);
-    writer.Null();
-  }
-
-  inline void operator()(const std::string& key, const char* value) {
-    writer.String(key);
-    writer.String(value);
-  }
-
-  inline void operator()(const std::string& key, const std::string& value) {
-    writer.String(key);
-    writer.String(value);
-  }
-
-  inline void operator()(const std::string& key, const double value) {
-    writer.String(key);
-    writer.Double(value);
-  }
-
-  inline void operator()(const std::string& key, const uint64_t value) {
-    writer.String(key);
-    writer.Uint64(value);
-  }
-
-  inline void operator()(const std::string& key, const int64_t value) {
-    writer.String(key);
-    writer.Int64(value);
-  }
-
-  inline void operator()(const std::string& key, const bool value) {
-    writer.String(key);
-    writer.Bool(value);
-  }
-
-  inline void operator()(const std::string& key, const std::nullptr_t) {
-    writer.String(key);
-    writer.Null();
-  }
-
-  inline void operator()(const char* value) {
-    writer.String(value);
-  }
-
-  inline void operator()(const std::string& value) {
-    writer.String(value);
-  }
-
-  inline void operator()(const double value) {
-    writer.Double(value);
-  }
-
-  inline void operator()(const uint64_t value) {
-    writer.Uint64(value);
-  }
-
-  inline void operator()(const int64_t value) {
-    writer.Int64(value);
-  }
-
-  inline void operator()(const bool value) {
-
-    writer.Bool(value);
-  }
-
-  inline void operator()(const std::nullptr_t) {
-    writer.Null();
+  template <typename K, typename V> inline void operator()(K key, V value) {
```

Comment:
Makes use of C++17's constexpr if, saves a few lines of code and more importantly gets rid of a lot of `static_cast`s from 32-bit to 64-bit integers scattered all around tyr